Gems/AWSCore/Code/Include/Framework/ServiceRequestJob.h

```diff
@@ -315,6 +301,26 @@ namespace AWSCore
         }
 
     private:
+        void SetAWSCredentials(IConfig* config)
+        {
+            if (RequestType::UseAWSCredentials())
+            {
+                if (!HasCredentials(config))
+                {
+                    m_missingCredentials = true;
+                    return;
+                }
+
+                m_AWSAuthSigner.reset(
+                    new Aws::Client::AWSAuthV4Signer{
+                        config->GetCredentialsProvider(),
+                        "execute-api",
+                        DetermineRegionFromRequestUrl()
+                    }
+                );
```

Comment:
minor: Instead of `return` if `!HasCredentials(config))`, just do the `m_AWSAuthSigner.reset` call in an else block. 
```suggestion
                if (!HasCredentials(config))
                {
                    m_missingCredentials = true;
                }
                else
                {
                    m_AWSAuthSigner.reset(
                        new Aws::Client::AWSAuthV4Signer{
                            config->GetCredentialsProvider(),
                            "execute-api",
                            DetermineRegionFromRequestUrl()
                        }
                    );
                }
```

Reply:
Dropped the change and reimplemented the fix.